--- a/bigtop-packages/src/common/hadoop/do-component-build	2017-06-01 16:21:26.000000000 -0500
+++ b/bigtop-packages/src/common/hadoop/do-component-build	2017-09-07 18:50:51.000000000 -0500
@@ -15,12 +15,18 @@
 # limitations under the License.

 set -ex
+
+# Snap builders create a proxy-aware m2 settings file; use it if present
+if [ -f $HOME/m2/settings.xml ] ; then
+  MAVEN_SETTINGS_OPT="-s $HOME/m2/settings.xml"
+fi
+
 ######################################################################
 # BIGTOP-2288 Please remove this patch when maven repository is updated
 # with version of leveldbjni-all library that is not x86 specific
 if [ $HOSTTYPE = "powerpc64le" ] ; then
         #download the POWER version of leveldb
-        git clone git://github.com/ibmsoe/leveldb.git
+        git clone https://github.com/ibmsoe/leveldb.git
         git clone https://github.com/ibmsoe/leveldbjni.git
         export SNAPPY_HOME=/usr/lib
         export LEVELDB_HOME=`cd leveldb; pwd`
@@ -34,7 +40,7 @@
         #Now use maven to build and update the local maven repository with ppc version of
         #leveldbjni.
         cd ${LEVELDBJNI_HOME}
-        mvn clean install -DskipTests -P download -Plinux64,all
+        mvn ${MAVEN_SETTINGS_OPT} clean install -DskipTests -P download -Plinux64,all
         cd ..
         #cleanup
         rm -rf ${LEVELDBJNI_HOME}
@@ -54,7 +60,7 @@
 mkdir build/src

 # Build artifacts
-MAVEN_OPTS="-Dzookeeper.version=$ZOOKEEPER_VERSION"
+MAVEN_OPTS="${MAVEN_SETTINGS_OPT} -Dzookeeper.version=$ZOOKEEPER_VERSION"
 MAVEN_OPTS="$MAVEN_OPTS -DskipTests -DskipTest -DskipITs"
 mvn $ANT_OPTS $BUNDLE_SNAPPY -Pdist -Pnative -Psrc -Dtar ${MAVEN_OPTS} -Dtomcat.version=${BIGTOP_TOMCAT_VERSION} -Dtomcat.download.url="http://archive.apache.org/dist/tomcat/tomcat-6/v${BIGTOP_TOMCAT_VERSION}/bin/apache-tomcat-${BIGTOP_TOMCAT_VERSION}.tar.gz" install "$@"
 mvn site site:stage ${MAVEN_OPTS} $@
