--- a/bigtop-packages/src/common/hadoop/install_hadoop.sh	2016-05-20 10:42:15.000000000 -0500
+++ b/bigtop-packages/src/common/hadoop/install_hadoop.sh	2017-09-06 11:29:52.000000000 -0500
@@ -265,26 +265,26 @@
 /sbin/modprobe fuse

 # Autodetect JAVA_HOME if not defined
-. /usr/lib/bigtop-utils/bigtop-detect-javahome
+. \${SNAP}/usr/lib/bigtop-utils/bigtop-detect-javahome

-export HADOOP_HOME=\${HADOOP_HOME:-${HADOOP_DIR#${PREFIX}}}
+export HADOOP_HOME=\${HADOOP_HOME:-\${SNAP}${HADOOP_DIR#${PREFIX}}}

-BIGTOP_DEFAULTS_DIR=\${BIGTOP_DEFAULTS_DIR-/etc/default}
+BIGTOP_DEFAULTS_DIR=\${BIGTOP_DEFAULTS_DIR-\${SNAP}/etc/default}
 [ -n "\${BIGTOP_DEFAULTS_DIR}" -a -r \${BIGTOP_DEFAULTS_DIR}/hadoop-fuse ] && . \${BIGTOP_DEFAULTS_DIR}/hadoop-fuse

-export HADOOP_LIBEXEC_DIR=${SYSTEM_LIBEXEC_DIR#${PREFIX}}
+export HADOOP_LIBEXEC_DIR=\${SNAP}${SYSTEM_LIBEXEC_DIR#${PREFIX}}

 if [ "\${LD_LIBRARY_PATH}" = "" ]; then
   export JAVA_NATIVE_LIBS="libjvm.so"
-  . /usr/lib/bigtop-utils/bigtop-detect-javalibs
-  export LD_LIBRARY_PATH=\${JAVA_NATIVE_PATH}:/usr/lib
+  . \${SNAP}/usr/lib/bigtop-utils/bigtop-detect-javalibs
+  export LD_LIBRARY_PATH=\${JAVA_NATIVE_PATH}:\${SNAP}/usr/lib:/usr/lib
 fi

 # Pulls all jars from hadoop client package
 for jar in \${HADOOP_HOME}/client/*.jar; do
   CLASSPATH+="\$jar:"
 done
-CLASSPATH="/etc/hadoop/conf:\${CLASSPATH}"
+CLASSPATH="\${SNAP_DATA}/etc/hadoop/conf:\${CLASSPATH}"

 env CLASSPATH="\${CLASSPATH}" \${HADOOP_HOME}/bin/fuse_dfs \$@
 EOF
@@ -354,7 +354,7 @@
 \
   <property>\
     <name>httpfs.hadoop.config.dir</name>\
-    <value>/etc/hadoop/conf</value>\
+    <value>${SNAP_DATA}/etc/hadoop/conf</value>\
   </property>' $HTTPFS_ETC_DIR/conf.empty/httpfs-site.xml

 # Make the pseudo-distributed config
@@ -375,9 +375,9 @@

 # FIXME: Provide a convenience link for configuration (HADOOP-7939)
 install -d -m 0755 ${HADOOP_DIR}/etc
-ln -s ${HADOOP_ETC_DIR##${PREFIX}}/conf ${HADOOP_DIR}/etc/hadoop
+ln -s ../../../..${HADOOP_ETC_DIR##${PREFIX}}/conf ${HADOOP_DIR}/etc/hadoop
 install -d -m 0755 ${YARN_DIR}/etc
-ln -s ${HADOOP_ETC_DIR##${PREFIX}}/conf ${YARN_DIR}/etc/hadoop
+ln -s ../../../..${HADOOP_ETC_DIR##${PREFIX}}/conf ${YARN_DIR}/etc/hadoop

 # Create log, var and lib
 install -d -m 0755 $PREFIX/var/{log,run,lib}/hadoop-hdfs
@@ -402,8 +402,8 @@
 for file in `cat ${BUILD_DIR}/hadoop-client.list` ; do
   for dir in ${HADOOP_DIR}/{lib,} ${HDFS_DIR}/{lib,} ${YARN_DIR}/{lib,} ${MAPREDUCE_DIR}/{lib,} ; do
     [ -e $dir/$file ] && \
-    ln -fs ${dir#$PREFIX}/$file ${CLIENT_DIR}/${file} && \
-    ln -fs ${dir#$PREFIX}/$file ${CLIENT_DIR}/${file/-[[:digit:]]*/.jar} && \
+    ln -fs ../../../..${dir#$PREFIX}/$file ${CLIENT_DIR}/${file} && \
+    ln -fs ../../../..${dir#$PREFIX}/$file ${CLIENT_DIR}/${file/-[[:digit:]]*/.jar} && \
     continue 2
   done
   exit 1
